### DISCUSSION TITLE
Creating a custom modal

### DISCUSSION THREAD

**Discussion by 0x7375** [gokcehan/lf#1699] [Q&A] [answered]
Is it possible to create a modal like the native "delete" one?
I currently have this trash command:

```
cmd trash %{{
    printf "trash? [y/N] "
    read ans
    if [[ -z "$ans" || "$ans" != "y" ]]; then
        lf -remote "send $id reload"
        return;
    fi

    files=$(printf "$fx" | tr '\n' ';')
    while [ "$files" ]; do
        file=${files%%;*}

        trash-put "$(basename "$file")"
        if [ "$files" = "$file" ]; then
            files=''
        else
            files="${files#*;}"
        fi
    done
    lf -remote "send $id reload"
}}
```
Which works fine but I would prefer not having to type enter, and pressing escape doesn't cancel the prompt

**Comment by DusanLesan** [comment, marked as answer]
@aayko I was playing with the flow proposed here: https://github.com/gokcehan/lf/discussions/1700
This is closest I could get so far:
```
map T &{{
	tmux split-window -l 1 '
		stty -icanon -echo
		echo -n "Trash files? [y/n] "
		key=$(dd bs=1 count=1 2>/dev/null)
		if [[ "$key" == "y" ]]; then
			while IFS= read -r file; do
				notify-send "Trashing file: $file"
			done <<< "'"$fx"'"
			lf -remote "send '$id' reload"
		fi
	'
}}
```
The solution requires you to have tmux installed, to know how to use it, and have `lf` started from tmux session like `tmux new-session 'lf'`. I could not find more elegant solution for taking single key that worked in tmux. 
This is how it looks:

https://github.com/gokcehan/lf/assets/20520590/a92ce97d-b145-4532-a518-0da86fc52dd4

**Reply by joelim-work** [reply]
Just adding a note that `%` prompts are line buffered (to allow editing) so you have to press <kbd>Enter</kbd> at some point. This is mentioned in the [documentation](https://github.com/gokcehan/lf/blob/master/doc.md#piping-shell-commands).

**Reply by 0x7375** [reply]
This is great thank you
